### ISSUE TITLE
Check blocks nested inside check blocks

### ISSUE THREAD

**Issue by jpolitz** [brownplt/pyret-lang#168]
It would be nice if this worked:

```
check:

   fun helper(type-of-set): ... end

  check "Testing with list-sets": helper(list-set) end
  check "Testing with tree-sets": helper(tree-set) end

end
```

But right now nested check blocks inside one another don't work.

**Comment by Emmay** [comment]
We tried this out:
```
fun add(n, a):
  n + a
end

check "A":
  add(8, 0) is 8
  add(9, 7) is 16
  fun double(n): n * 2 end
  check "B":
    double(4) is 8
  end
  add(double(2), 5) is 9
end
```
And got 4 passing tests, however they were all grouped under the "A" block. Is this sufficient to close, or should the title of this issue change to emphasize what doesn't work?

**Comment by blerner** [comment]
No, I think that's exactly what Joe doesn't want to have happen.  Why bother naming the check block, if it doesn't get shown and used for organization?

**Comment by Emmay** [comment]
OK, does that mean the title of this issue should be changed? I was under the impression that "don't work" means the tests would generate an error or not get called at all, but the actual issue is that the results of tests do not reference the nested check blocks by name, correct?
